docs/public_ips.md

```diff
@@ -0,0 +1,289 @@
+# Load balancing & publicly accessible services
+
+This document describes how kubernetes services are made accessible outside the cluster.
+
+There are two main mechanisms:
+
+* The service can be exposed through a load-balancer managed by k8s.
+* k8s can also expose services without an associated load-balancer; the service could be reached via DNS SRV records,
+or by a load-balancer configured by a helper process, or even by a client using the k8s API directly for discovery.
+_(new functionality)_
+
+## Public ports
+
+To be exposed outside the k8s cluster, the service must be on known ports on externally reachable
+IPs.  For version 1 public services, k8s exposes each public service port on a common port on each
+node.
+
+## Important scenarios
+
+The names in bold are the names we will use in the API.
+
+1. **cluster** The service is reachable from within the cluster on a virtual IP.  _(Existing functionality)_
+1. **public** The service is exposed on a set of nodes, on a set of ports.  _(New functionality)_.
+ For V1 services are exposed on all nodes.  The service endpoints would be discoverable via the k8s API,
+ and helper processes could then expose that information via DNS or configure external systems
+ (e.g. physical load balancers)  The intention is that this service should then be reachable from outside
+ the k8s cluster; a cloud-provider would automatically configure firewall rules (or equivalents)
+ so that a public service is reachable from the public Internet.
+1. **loadbalancer** The service is exposed via a k8s-managed load-balancer.  _(Existing functionality)_
+
+Expected future functionality:
+
+1. **namespace** The service is only accessible to pods belonging to the same namespace.  This is similar
+to _cluster_, but with additional firewall rules.
+1. **dmz** The service is accessible to pods that are in the DMZ.  This is similar to _cluster_,
```

Comment:
Could someone point me to prior discussion on dmz? I don't understand why this isn't namespace-scoped visibility, plus another mechanism to restrict access to services outside the namespace, which seems like something that belongs in SecurityContext.